### PR TITLE
Ports the XprojProjectFactory from the DotNet project system

### PR DESCRIPTION
**Pull request by 333fred**
This simply blocks loading of any Xproj templates that attempt to load, which is how the functionality will ship in Preview 5. We will enable migration in an OOB.

Tagging @BillHiebert @davkean for review. /cc @dotnet/project-system

### CODE REVIEW

**Comment by 333fred** [comment]
This is the start of the implementation work around #213

File: src/Microsoft.VisualStudio.ProjectSystem.CSharp.VS/ProjectSystem/VS/XprojProjectFactory.cs [hunk 1]
```diff
@@ -0,0 +1,155 @@
+//--------------------------------------------------------------------------------------------
+// Implements a VS project factory for 
+//
+// Copyright(c) 2016 Microsoft Corporation
+//--------------------------------------------------------------------------------------------
```
Comment (abpiskunov) [inline src/Microsoft.VisualStudio.ProjectSystem.CSharp.VS/ProjectSystem/VS/XprojProjectFactory.cs]: header

File: src/Microsoft.VisualStudio.ProjectSystem.CSharp.VS/ProjectSystem/VS/XprojProjectFactory.cs [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+using System;
+using System.ComponentModel.Composition;
+using System.Runtime.InteropServices;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+using Microsoft.VisualStudio.Packaging;
+
+namespace Microsoft.VisualStudio.ProjectSystem.DotNet.VisualStudio
```
Comment (abpiskunov) [inline src/Microsoft.VisualStudio.ProjectSystem.CSharp.VS/ProjectSystem/VS/XprojProjectFactory.cs]: >DotNet [](start = 47, length = 6)

namespace?

File: src/Microsoft.VisualStudio.ProjectSystem.CSharp.VS/ProjectSystem/VS/XprojProjectFactory.cs (hunk 1 above)
Comment (333fred) [inline src/Microsoft.VisualStudio.ProjectSystem.CSharp.VS/ProjectSystem/VS/XprojProjectFactory.cs]: Whoops, I could have sworn I fixed those two, but I guess I didn't.

**Comment by 333fred** [comment]
Ping?

**Comment by srivatsn** [comment]
:shipit:

**Pull request merged**